common/remote/rmtsmtp.cpp

```diff
@@ -633,6 +649,70 @@ class CMailInfo
         out.append("HELO ").append(mailServer.get()).append("\r\n");
     }
 
+    void getEhlo(StringBuffer & out) const
+    {
+        out.append("EHLO ").append(mailServer.get()).append("\r\n");
+    }
+
+    void getEhloAndOptions(StringBuffer & out)
+    {
+        // Initial EHLO and gather reply, which should contain ESMTP options
+        getEhlo(out);
+        writeAndAck(out.str(), out.length());
+
+#if defined(_USE_OPENSSL)
+        bool useTLS = false;
+        if (inlen > 0)
+        {
+            // Walk the buffer, looking for the STARTTLS option
+            char* buffPtr = inbuff;
+            while (buffPtr < (inbuff + inlen))
+            {
+                if (strncmp(buffPtr, "250", 3) == 0)
```

Comment:
Could the "250" occur not at the beginning of buffPtr ?
And if in this block, could buffPtr go past inbuff + inlen ?

Reply:
My understanding is that there will always be at least one 250 response that serves as an acknowledgement of the EHLO. In my tests, that line was "250-Nice to meet you." but the exact wording is not dictated.

Response codes will always be at the beginning of the buffer, and 250 responses should be first if there is no error.

Reply:
I modified the size of the outer while() to abort earlier to avoid some scans outside of the buffer.

Reply:
I am not sure of the reason for a while loop at the beginning ?